src/builders.jl

```diff
@@ -62,6 +62,42 @@ function build(builder::Short, rng, n, m)
         Flux.Dense(n_hidden, m, init=init))
 end
 
+"""
+    MLP(; hidden=(10,), σ=Flux.sigmoid, rng=GLOBAL_RNG)
+
+MLJFlux builder that constructs a Multi-layer perceptron network. The ith element of `hidden` represents the number of neurons in the ith hidden layer. An activation function `σ` is applied between each layer.
+
+The each layer is initialized using `Flux.glorot_uniform(rng)`. If
+`rng` is an integer, it is instead used as the seed for a
+`MersenneTwister`.
+
+"""
+mutable struct MLP{N} <: MLJFlux.Builder
+    hidden::NTuple{N, Int}  # count first and last layer
+    σ
+end
+MLP(; hidden=(10,), σ=Flux.sigmoid) = MLP(hidden, σ)
+function MLJFlux.build(mlp::MLP, rng, n_in, n_out)
+    init=Flux.glorot_uniform(rng)
+    hidden = []
+    for i ∈ 2:length(mlp.hidden)
+        push!(hidden, Flux.Dense(mlp.hidden[i-1], mlp.hidden[i], mlp.σ, init=init))
+    end
+
+    return Flux.Chain(
+        Flux.Dense(n_in, mlp.hidden[1], mlp.σ, init=init),
+        hidden...,
+        Flux.Dense(mlp.hidden[end], n_out, init=init)
```

Comment:
The first and last layers can probably be included in the definition of/`push!`ed into `hidden` directly. Then the splat can cover everything.

Reply:
That makes sense. In my head I was keeping the input/output layers separate from the hidden layers.